rfcs/0004-page-transitions.md

```diff
@@ -0,0 +1,207 @@
+- Start Date: 2020-01-13
+- Project: [next-with-moxy](https://github.com/moxystudio/next-with-moxy)
+- Reference Issues: N/A
+- Implementation PR: N/A
+
+# Summary
+
+This proposal is to add full page transitions to the `next-with-moxy` boilerplate, either as a guide and easy to use feature (recipe) or an out-of-the-box feature (by including it in the source code).
+
+# Basic example
+
+We will use [Highway](https://highway.js.org/) by Dogstudio, a lightweight library to create the navigation animations.
+
+## Installation
+
+First, install the package:
+
+```sh
+$ npm install --save @dogstudio/highway
+```
+
+## Enabling Highway
+
+In our [`App.js`](https://github.com/moxystudio/next-with-moxy/blob/master/www/app/App.js) file, we'll have to enable Highway:
+
+```js
+import Highway from '@dogstudio/highway/src/highway';
+import { Slide, Fade } from '../shared/components/transitions';
+
+class App extends NextApp {
+    Highway;
+
+    componentDidMount() {
+        this.unregisterGoogleTracking = registerGoogleTracking(this.props.router);
+
+        this.Highway = new Highway.Core({
```

Comment:
Do we need to keep a reference to it? Perhaps we need it to destroy in `componentWillUnmount`? If not, I would get rid of it.

Reply:
In this example we don't need to keep the reference. I just kept it there to show that in some cases we may need it, for example with [events](https://highway.js.org/examples/events.html) probably.